### ISSUE TITLE
Successful install, Failed Download (Soundcloud) Traceback.. Object has no attribute "replace"

### ISSUE THREAD

**Issue by tmanok** [Miserlou/SoundScrape#226]
Hi SoundScrape!

I successfully installed SoundScrape on Mac OS Mojave using Pip 19.newest.whatever and commands specific to Soundscrape function fine, but the actual process of downloading SoundCloud fails completely. Now before I show you what my original error was, let me explain that I noticed I was running python 2.7- by the way I hate python for this very reason, it's the messiest thing I've ever dealt with in any package management system but anyway while running PY2.7 on Brew:

OUTPUT:
```
MegaServer:hypomaniacs Tmonster$ sudo soundscrape hypomaniacs
Traceback (most recent call last):
  File "/usr/local/bin/soundscrape", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/soundscrape/soundscrape.py", line 119, in main
    process_soundcloud(vargs)
  File "/usr/local/lib/python2.7/site-packages/soundscrape/soundscrape.py", line 292, in process_soundcloud
    id3_extras=id3_extras)
  File "/usr/local/lib/python2.7/site-packages/soundscrape/soundscrape.py", line 460, in download_tracks
    puts_safe(e)
  File "/usr/local/lib/python2.7/site-packages/soundscrape/soundscrape.py", line 1315, in puts_safe
    puts(text)
  File "/usr/local/lib/python2.7/site-packages/clint/textui/core.py", line 57, in puts
    s = tsplit(s, NEWLINES)
  File "/usr/local/lib/python2.7/site-packages/clint/utils.py", line 69, in tsplit
    string = string.replace(i, final_delimiter)
AttributeError: 'HTTPError' object has no attribute 'replace'
```
After noticing my apparent problem I decided to clean up my machine by installing Python 3, which was actually already installed a while back but somehow never ended up being used by everything. 

```
brew uninstall --ignore-dependencies python
brew uninstall --ignore-dependencies python2
brew remove python@2
brew uninstall --ignore-dependencies python3
brew install python
brew unlink python && brew link python
brew unlink python3 && brew link python3
brew doctor
brew prune
brew upgrade
```
Then I had to recreate my simlinks with pip by reinstalling pip3 and then I had to reinstall soundscrape (basically tell pip3 that it was there).

```
curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
sudo python get-pip.py
sudo pip install soundscrape
```
All done right? Not quite, much bigger error than before. Yes I did test my other pip and python software, which is operating very well surprisingly after all that kerfuffle. 

```
Downloading: Cyberpunkers - Shock (Hypomaniacs RMX)
Problem downloading Cyberpunkers - Shock (Hypomaniacs RMX)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/soundscrape/soundscrape.py", line 437, in download_tracks
    stream = client.get(track['stream_url'], allow_redirects=False, limit=200)
  File "/usr/local/lib/python3.7/site-packages/soundcloud/client.py", line 133, in _request
    return wrapped_resource(make_request(method, url, kwargs))
  File "/usr/local/lib/python3.7/site-packages/soundcloud/request.py", line 148, in make_request
    result.raise_for_status()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 429 Client Error: Unknown for url: https://api.soundcloud.com/tracks/168294290/stream?limit=200&client_id=175c043157ffae2c6d5fed16c3d95a4c

During handling of the above exception, another exception occurred:

MegaServer:hypomaniacs Tmonster$ sudo soundscrape hypomaniacs
Traceback (most recent call last):
  File "/usr/local/bin/soundscrape", line 11, in <module>
    load_entry_point('soundscrape==0.30.1', 'console_scripts', 'soundscrape')()
  File "/usr/local/lib/python3.7/site-packages/soundscrape/soundscrape.py", line 119, in main
    process_soundcloud(vargs)
  File "/usr/local/lib/python3.7/site-packages/soundscrape/soundscrape.py", line 292, in process_soundcloud
    id3_extras=id3_extras)
  File "/usr/local/lib/python3.7/site-packages/soundscrape/soundscrape.py", line 460, in download_tracks
    puts_safe(e)
  File "/usr/local/lib/python3.7/site-packages/soundscrape/soundscrape.py", line 1315, in puts_safe
    puts(text)
  File "/usr/local/lib/python3.7/site-packages/clint/textui/core.py", line 57, in puts
    s = tsplit(s, NEWLINES)
  File "/usr/local/lib/python3.7/site-packages/clint/utils.py", line 69, in tsplit
    string = string.replace(i, final_delimiter)
AttributeError: 'HTTPError' object has no attribute 'replace'
```
Thanks for all your help. Let me know when it's working again! :)

**Comment by vStone** [comment]
This might be related to the rate limits soundcloud has imposed:

```
Effective July 1, all requests that result in access to a playable stream are subject to a limit of 15,000 requests per any 24-hour time window. Currently, this is inclusive of all requests that are made to /tracks/:id/stream.
```

The api is giving me the following json as result:
```
{"errors":[{"meta":{"rate_limit":{"bucket":"by-client","max_nr_of_requests":15000,"time_window":"PT24H","name":"plays"},"remaining_requests":0,"reset_time":"2019/03/11 15:43:10 +0000"}}]}
```

**Comment by deweydb** [comment]
this PR fixes it: https://github.com/Miserlou/SoundScrape/pull/205/commits/87dd035e95763cd84be9617b38c0f8a6e0aa3e17

**Comment by SimplicityGuy** [comment]
There's lot of issues with how http errors were not handled here. This is fixed in #241.
